docs/content/preview/releases/techadvisories/ta-23476.md

```diff
@@ -17,161 +17,105 @@ type: docs
+If you have already inserted data with non-zero microsecond precision, contact Yugabyte support for assistance running a script to read and re-insert the affected rows with millisecond precision.
 
 ## Details
 
-The YCQL wire protocol uses millisecond precision for timestamps, while the YCQL storage system uses microsecond precision internally. This means that all timestamps that can be inserted are expected to have `000` for the microseconds component.
+Cassandra (the base for YCQL) only supports timestamps with millisecond precision. However, YCQL's underlying storage can store timestamps with microsecond precision, leading to inconsistencies. When timestamps with non-zero microseconds were inserted into YCQL, they were stored in the underlying storage with microsecond precision. However, when queried, the Cassandra layer rounded these timestamps to milliseconds, losing the microsecond details.
 
-However, the `currenttimestamp()` function currently generates values with microsecond precision, so the microseconds component is likely to be non-zero 99.9% of the time.
+As a result, users encountered issues such as:
 
-When these non-zero microsecond timestamp values are stored, the microsecond information will be lost when the precision is adjusted to milliseconds for the wire protocol (for example, when responding to a user query). As a result, you will always see timestamp values with only millisecond precision.
-
-This can create issues in certain scenarios as follows:
-
-- Equality conditions: If you query for a timestamp value that was just read, the condition will not match, as the microsecond component will differ (0 for the query, non-zero for the stored value).
-
-- Index scans: If the primary key of the main table includes an affected timestamp column, the data in both the table and the index will be in microsecond precision, so they will match. However, when performing an index scan, the main-table read request uses the same protocol as the YCQL API, which adjusts the precision to milliseconds. This may cause the main-table read request to not match, and the index scan result to miss the affected rows with non-zero microsecond values.
+1. **Fetch Mismatch**: When querying rows inserted using `current timestamp()` or explicit timestamp strings, the `SELECT *` would return timestamps rounded to milliseconds. If the same value was then used in a WHERE clause to filter data, it would fail to fetch the expected rows.
+1. **Index Scan Failure**: For tables with a timestamp as part of the primary key and a secondary index, index scans would fail. The secondary index would retrieve the timestamp stored in milliseconds, but when fetching the full table data from underlying storage (where it was stored in microseconds), the values would not match.
```

Comment:
```suggestion
- **Index scan failure**: For tables with a timestamp as part of the primary key and a secondary index, index scans would fail. The secondary index would retrieve the timestamp stored in milliseconds, but when fetching the full table data from underlying storage (where it was stored in microseconds), the values would not match.
```